diff -urN build.o/package/libupnpp/001-ff.patch build/package/libupnpp/001-ff.patch
--- build.o/package/libupnpp/001-ff.patch	1970-01-01 01:00:00.000000000 +0100
+++ build/package/libupnpp/001-ff.patch	2019-03-10 23:12:46.847341000 +0100
@@ -0,0 +1,12 @@
+diff -urN libupnpp/libupnpp/control/cdirectory.cxx build/libupnpp/control/cdirectory.cxx
+--- libupnpp/libupnpp/control/cdirectory.cxx	2017-01-26 11:34:14.655347460 +0100
++++ build/libupnpp/control/cdirectory.cxx	2017-01-26 10:18:11.757439404 +0100
+@@ -46,6 +46,8 @@
+ using namespace std::placeholders;
+ using namespace UPnPP;
+ 
++extern "C" { int atoi (const char *s);}
++
+ namespace UPnPClient {
+ 
+ // The service type string for Content Directories:
diff -urN build.o/package/libupnpp/libupnpp.hash build/package/libupnpp/libupnpp.hash
--- build.o/package/libupnpp/libupnpp.hash	2016-05-31 23:52:36.000000000 +0200
+++ build/package/libupnpp/libupnpp.hash	2019-03-10 23:15:38.759341000 +0100
@@ -1,2 +1 @@
-# Hashes from: http://www.lesbonscomptes.com/upmpdcli/downloads/libupnpp-0.14.1.tar.gz.sha256
-sha256  ffb33379f3734419de683b0adca5228e802c64d3960681c15f3a8de7d7e315d1  libupnpp-0.14.1.tar.gz
+sha256 d2ab5a4bb6df035ddddd11542bd4631ab7031354259145f07fbc3c56a8f50fac  libupnpp-0.16.1.tar.gz
diff -urN build.o/package/libupnpp/libupnpp.mk build/package/libupnpp/libupnpp.mk
--- build.o/package/libupnpp/libupnpp.mk	2016-05-31 23:52:36.000000000 +0200
+++ build/package/libupnpp/libupnpp.mk	2019-03-10 23:13:50.279341000 +0100
@@ -4,7 +4,7 @@
 #
 ################################################################################
 
-LIBUPNPP_VERSION = 0.14.1
+LIBUPNPP_VERSION = 0.16.1
 LIBUPNPP_SITE = http://www.lesbonscomptes.com/upmpdcli/downloads
 LIBUPNPP_LICENSE = GPLv2+
 LIBUPNPP_LICENSE_FILES = COPYING
