From 499b499cf5835f2995189f40a7f785b9a21d57b7 Mon Sep 17 00:00:00 2001
From: Felix Schmidt <fesc2000@gmail.com>
Date: Wed, 23 Sep 2020 20:41:32 +0200
Subject: [PATCH] Fix makefile

---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index bea5d3c..81f8974 100644
--- a/Makefile
+++ b/Makefile
@@ -49,8 +49,8 @@
 
 # Configuration
 #
-CC=$(CROSS_COMPILE)gcc
-CFLAGS=$(CASH_CFLAGS) -Wall -Wno-unknown-pragmas -Wformat=0
+CC	?= $(CROSS_COMPILE)gcc
+CFLAGS   = $(CASH_CFLAGS) -Wall -Wno-unknown-pragmas -Wformat=0
 
 PWD=$(shell pwd)
 
@@ -199,8 +199,8 @@ clean:
 $(OBJDIR)/libcash.a:	$(DYNCALL_LIB) ${BUILD_OBJS} 
 	${CROSS_COMPILE}ar crus $@ ${BUILD_OBJS} $(shell ls $(DYNCALL_DIR)/dyncall/*.o)
 
-$(OBJDIR)/cash_wrap.so:	${BUILD_SHOBJS} $(shell ls $(DYNCALL_DIR)/dyncall/*.o)
-	${CROSS_COMPILE}gcc -fPIC -shared -o $@ $^ ${LDFLAGS} ${LIBS}  -pthread -lm 
+$(OBJDIR)/cash_wrap.so:	${BUILD_SHOBJS} $(DYNCALL_LIB)
+	$(CC) -fPIC -shared -o $@ $^ ${LDFLAGS} ${LIBS}  -pthread -lm 
 	cp $@ .
 
 ifeq ($(PREBUILT_YACC_SRC),1)
-- 
2.20.1

