--- orig/etc/udev/rules.d/50-udev-default.rules	2016-10-27 12:51:38.000000000 +0200
+++ squashfs-root/etc/udev/rules.d/50-udev-default.rules	2017-01-20 10:39:47.700544958 +0100
@@ -5,6 +5,7 @@
 KERNEL=="ptmx",			GROUP="tty", MODE="0666"
 KERNEL=="tty",			GROUP="tty", MODE="0666"
 KERNEL=="tty[0-9]*",		GROUP="tty", MODE="0620"
+KERNEL=="ttyACM[0-9]*",		GROUP="tty", MODE="0666"
 KERNEL=="vcs|vcs[0-9]*|vcsa|vcsa[0-9]*", GROUP="tty"
 
 # serial
