diff --git a/audio_pipe.c b/audio_pipe.c
index 4ec9b24..2143678 100644
--- a/audio_pipe.c
+++ b/audio_pipe.c
@@ -42,8 +42,36 @@
 static int fd = -1;
 
 char *pipename = NULL;
+static char *volname = NULL;
 int warned = 0;
 
+static double minDb = (double)-64.0;
+static double maxDb = (double)0.0;
+
+extern audio_output audio_pipe;
+
+static void volume(double vol)
+{
+    FILE *fh;
+    double nvol = ((vol - minDb) / (maxDb - minDb)) * (double)100.0;
+
+    if (!volname)
+	return;
+
+    fh = fopen(volname, "w");
+    if (!fh)
+	return;
+
+    fprintf (fh, "%d", (int)nvol);
+
+    fclose (fh);
+}
+
+static void parameters(audio_parameters *info) {
+  info->minimum_volume_dB = (int)minDb;
+  info->maximum_volume_dB = (int)maxDb;
+}
+
 static void start(__attribute__((unused)) int sample_rate,
                   __attribute__((unused)) int sample_format) {
 
@@ -113,6 +141,12 @@ static int init(int argc, char **argv) {
 
     if ((pipename) && (strcasecmp(pipename, "STDOUT") == 0))
       die("Can't use \"pipe\" backend for STDOUT. Use the \"stdout\" backend instead.");
+
+    if (config_lookup_string(config.cfg, "pipe.volume", &str)) {
+      volname = (char *)str;
+      audio_pipe.volume = &volume;
+      audio_pipe.parameters = &parameters;
+    }
   }
 
   if ((pipename == NULL) && (argc != 1))
@@ -149,5 +183,5 @@ audio_output audio_pipe = {.name = "pipe",
                            .delay = NULL,
                            .play = &play,
                            .volume = NULL,
-                           .parameters = NULL,
+                           .parameters = &parameters,
                            .mute = NULL};
