--- tvheadend-221c29b40b1e53ae09a69d9458442dd4fea665f5_org/src/main.c	2020-04-13 10:14:10.097503197 -0400
+++ tvheadend-221c29b40b1e53ae09a69d9458442dd4fea665f5/src/main.c	2020-04-13 10:15:42.709275802 -0400
@@ -1204,7 +1204,7 @@
     rl.rlim_cur = 2*1024*1024;
     if (setrlimit(RLIMIT_STACK, &rl)) {
       tvhlog(LOG_ALERT, LS_START, "too small stack size - %ld", (long)rl2);
-      return 1;
+//      return 1;
     }
   }
 
