--- orig/etc/init.d/rc.tail.sh	2017-05-05 12:44:34.000000000 +0000
+++ squashfs-root/etc/init.d/rc.tail.sh	2017-05-11 08:18:53.091167577 +0000
@@ -46,5 +46,56 @@
 set_m_sleep=$((10*60))
 nohup sh -c "echo \"\$0[\$\$]: ++++fork set_modulemen, sleep ${set_m_sleep}++++\" > /dev/console ; sleep ${set_m_sleep}; echo \"\$0[\$\$]: ++++do set_modulemen++++\" > /dev/console; /bin/set_modulemem;" &
 fi
+
 #########################################################################
+# MOD: Make sure telnetd is started after startup for 5 mins
+#	Do this only if /sbin/ar7login exists. If not then this is 
+#	Likely firmware >= 6.83 where we don't need it.
+#########################################################################
+if [ -f /sbin/ar7login ]; then
+	/usr/sbin/telnetd -l /sbin/ar7login
+	nohup sh -c "sleep 300; echo killing telnetd > /dev/console; sleep 10; killall telnetd" &
+fi
+
+#########################################################################
+# MOD: Start a SSH daemon (dropbear)
+#   - We want to be able to store root password persistently:
+#	- Create a copy for /etc/shadow to /nvram/shadow (if not yet done)
+#	- Link /etc/shadow to /nvram/shadow
+#   - Same for the certificates:
+#	- Create /nvram/dropbear (if it does not yet exist)
+#	- Link /var/tmp/dropbear to /nvram/dropbear
+#   - Same for roots .ssh directory (var/tmp/root-ssh linked to /nvram/root-ssh)
+#   - Start the daemon. -R will create certificates on demand
+#   
+#########################################################################
+if [ -x /usr/local/sbin/dropbear ]; then
+    if [ ! -f /nvram/shadow ]; then
+	cp /etc/shadow /nvram/shadow
+	chmod 600 /nvram/shadow
+    fi
+    rm -f /var/tmp/shadow
+    ln -sf /nvram/shadow /var/tmp/shadow
+
+    if [ ! -d /nvram/dropbear ]; then
+	if [ -d /var/media/ftp/.dropbear ]; then
+		mv /var/media/ftp/.dropbear /nvram/dropbear
+	else
+		mkdir /nvram/dropbear
+		chmod 700 /nvram/dropbear
+	fi
+    fi
+    rm -rf /var/tmp/dropbear
+    ln -sf /nvram/dropbear /var/tmp/dropbear
+
+    if [ ! -d /nvram/root-ssh ]; then
+	mkdir /nvram/root-ssh
+	chmod 700 /nvram/root-ssh
+    fi
+    rm -rf /var/tmp/root-ssh
+    ln -sf /nvram/root-ssh /var/tmp/root-ssh
+
+    /usr/local/sbin/dropbear -R
+fi
+
 exit 0
