--- orig/etc/init.d/rc.tail.sh	2016-03-02 19:39:24.000000000 +0000
+++ mod/etc/init.d/rc.tail.sh	2016-09-04 09:27:43.240275632 +0000
@@ -47,4 +47,49 @@
 nohup sh -c "echo \"\$0[\$\$]: ++++fork set_modulemen, sleep ${set_m_sleep}++++\" > /dev/console ; sleep ${set_m_sleep}; echo \"\$0[\$\$]: ++++do set_modulemen++++\" > /dev/console; /bin/set_modulemem;" &
 fi
 #########################################################################
+# MOD: Make sure telnetd is started
+#########################################################################
+/usr/sbin/telnetd -l /sbin/ar7login
+
+#########################################################################
+# MOD: Start a SSH daemon (dropbear)
+#   - We want to be able to store root password persistently:
+#	- Create a copy for /etc/shadow to /nvram/shadow (if not yet done)
+#	- Link /etc/shadow to /nvram/shadow
+#   - Same for the certificates:
+#	- Create /var/media/ftp/.dropbear (if it does not yet exist)
+#	- Link /var/tmp/dropbear to /var/media/ftp/.dropbear
+#   - Start the daemon. -R will create certificates on demand
+#
+#   TODO: use box password for root, or add GUI to edit ssh pwd
+#   
+#########################################################################
+if [ -x /usr/local/sbin/dropbear ]; then
+    if [ ! -f /nvram/shadow ]; then
+	cp /etc/shadow /nvram/shadow
+    fi
+    rm -f /var/tmp/shadow
+    ln -sf /nvram/shadow /var/tmp/shadow
+
+    if [ ! -d /var/media/ftp/.dropbear ]; then
+	mkdir /var/media/ftp/.dropbear
+	chmod 700 /var/media/ftp/.dropbear
+    fi
+
+    rm -rf /var/tmp/dropbear
+    ln -sf /var/media/ftp/.dropbear /var/tmp/dropbear
+
+    /usr/local/sbin/dropbear -R
+fi
+
+#########################################################################
+# MOD: telnetd also on atom core
+#########################################################################
+/usr/sbin/start_atom_telnetd
+
+#########################################################################
+# MOD: call debug.cfg, if present
+#########################################################################
+. /var/flash/debug.cfg
+
 exit 0
