diff --git a/example/a2dp_sink_demo.c b/example/a2dp_sink_demo.c
index 7d4cb21..eace88c 100644
--- a/example/a2dp_sink_demo.c
+++ b/example/a2dp_sink_demo.c
@@ -86,7 +86,7 @@
 
 #ifdef HAVE_POSIX_FILE_IO
 #include "wav_util.h"
-#define STORE_SBC_TO_SBC_FILE 
+#undef STORE_SBC_TO_SBC_FILE 
 #define STORE_SBC_TO_WAV_FILE 
 #endif
 
@@ -151,6 +151,8 @@ static FILE * sbc_file;
 static char * sbc_filename = "avdtp_sink.sbc";
 #endif
 
+const char *sink_name = "A2DP Sink Demo 00:00:00:00:00:00";
+
 typedef struct {
     // bitmaps
     uint8_t sampling_frequency_bitmap;
@@ -290,7 +292,7 @@ static int a2dp_sink_and_avrcp_services_init(void){
     // Set local name with a template Bluetooth address, that will be automatically
     // replaced with a actual address once it is available, i.e. when BTstack boots
     // up and starts talking to a Bluetooth module.
-    gap_set_local_name("A2DP Sink Demo 00:00:00:00:00:00");
+    gap_set_local_name(sink_name);
     gap_discoverable_control(1);
     gap_set_class_of_device(0x200408);
 
@@ -1131,6 +1133,17 @@ int btstack_main(int argc, const char * argv[]){
     (void)argc;
     (void)argv;
 
+    if (argc > 1)
+	sink_name = argv[1];
+
+    if (argc > 2)
+	wav_filename = (char*)argv[2];
+
+    printf ("name: %s  output=%s\n", sink_name, wav_filename);
+ 
+	
+
+
     int err = a2dp_sink_and_avrcp_services_init();
     if (err) return err;
     // turn on!
diff --git a/platform/libusb/hci_transport_h2_libusb.c b/platform/libusb/hci_transport_h2_libusb.c
index 23c73eb..df46dbb 100644
--- a/platform/libusb/hci_transport_h2_libusb.c
+++ b/platform/libusb/hci_transport_h2_libusb.c
@@ -77,6 +77,14 @@
 
 #define ASYNC_POLLING_INTERVAL_MS 1
 
+#ifndef LIBUSB_LOG_LEVEL_WARNING
+#define LIBUSB_LOG_LEVEL_WARNING 1
+#endif
+
+#ifndef LIBUSB_LOG_LEVEL_ERROR
+#define LIBUSB_LOG_LEVEL_ERROR 2
+#endif
+
 //
 // Bluetooth USB Transport Alternate Settings:
 //
diff --git a/platform/posix/btstack_link_key_db_fs.c b/platform/posix/btstack_link_key_db_fs.c
index ade3c52..96a0ab5 100644
--- a/platform/posix/btstack_link_key_db_fs.c
+++ b/platform/posix/btstack_link_key_db_fs.c
@@ -54,7 +54,7 @@
 #ifdef _WIN32
 #define LINK_KEY_PATH ""
 #else
-#define LINK_KEY_PATH "/tmp/"
+#define LINK_KEY_PATH "/var/run/bt/"
 #endif
 #endif
 
diff --git a/port/libusb/Makefile b/port/libusb/Makefile
index 67c28fb..e9411c7 100644
--- a/port/libusb/Makefile
+++ b/port/libusb/Makefile
@@ -1,3 +1,9 @@
+BUILDROOT   = $(shell cd ../../../../buildroot/build; pwd)
+SYSROOT     = $(BUILDROOT)/output/host/usr/i686-buildroot-linux-uclibc/sysroot
+TOOLCHAIN   = $(shell cd $(BUILDROOT)/output/host/usr/bin/; pwd)
+export PATH := $(TOOLCHAIN):$(PATH)
+CC=i686-buildroot-linux-uclibc-gcc
+
 # Makefile for libusb based examples
 BTSTACK_ROOT = ../..
 
@@ -20,8 +26,10 @@ VPATH += ${BTSTACK_ROOT}/platform/libusb
 VPATH += ${BTSTACK_ROOT}/chipset/csr
 
 # use pkg-config
-CFLAGS  += $(shell pkg-config libusb-1.0 --cflags)
-LDFLAGS += $(shell pkg-config libusb-1.0 --libs)
+#CFLAGS  += $(shell pkg-config libusb-1.0 --cflags)
+#LDFLAGS += $(shell pkg-config libusb-1.0 --libs)
+CFLAGS  += -I$(SYSROOT)/usr/include/libusb-1.0
+LDFLAGS += -L$(SYSROOT)/usr/lib -lusb-1.0
 
 EXAMPLES += csr_set_bd_addr
 
diff --git a/port/libusb/btstack_config.h b/port/libusb/btstack_config.h
index 7056a82..fd09945 100644
--- a/port/libusb/btstack_config.h
+++ b/port/libusb/btstack_config.h
@@ -8,7 +8,7 @@
 // Port related features
 #define HAVE_MALLOC
 #define HAVE_POSIX_FILE_IO
-#define HAVE_BTSTACK_STDIN
+#undef HAVE_BTSTACK_STDIN
 #define HAVE_POSIX_TIME
 
 // BTstack features that can be enabled
