--- orig/etc/init.d/rc.tail.sh	2017-05-05 12:43:17.000000000 +0000
+++ squashfs-root/etc/init.d/rc.tail.sh	2017-05-11 08:24:09.566755199 +0000
@@ -47,4 +47,15 @@
 nohup sh -c "echo \"\$0[\$\$]: ++++fork set_modulemen, sleep ${set_m_sleep}++++\" > /dev/console ; sleep ${set_m_sleep}; echo \"\$0[\$\$]: ++++do set_modulemen++++\" > /dev/console; /bin/set_modulemem;" &
 fi
 #########################################################################
+
+#########################################################################
+# MOD: Make sure telnetd is started after startup for 5 mins
+#########################################################################
+/usr/local/bin/telnetd -l /sbin/ar7login
+nohup sh -c "sleep 300; echo killing telnetd > /dev/console; sleep 10; killall telnetd" &
+
+# ffritz services startup
+#
+/usr/local/etc/init.d/ff_atom_startup
+
 exit 0
