diff -urN libupnpp/libupnpp/control/cdirectory.cxx build/libupnpp/control/cdirectory.cxx
--- libupnpp/libupnpp/control/cdirectory.cxx	2017-01-26 11:34:14.655347460 +0100
+++ build/libupnpp/control/cdirectory.cxx	2017-01-26 10:18:11.757439404 +0100
@@ -46,6 +46,8 @@
 using namespace std::placeholders;
 using namespace UPnPP;
 
+extern "C" { int atoi (const char *s);}
+
 namespace UPnPClient {
 
 // The service type string for Content Directories:
